src/main/java/com/powsybl/openloadflow/network/impl/Transformers.java

```diff
@@ -25,238 +23,153 @@
+        } else if (ptc != null && rtc != null) {
+            Objects.requireNonNull(rtcPosition);
+            Objects.requireNonNull(ptcPosition);
+            RatioTapChangerStep rtcStep = rtc.getStep(rtcPosition);
+            PhaseTapChangerStep ptcStep = ptc.getStep(ptcPosition);
+            r *= (1 + rtcStep.getR() / 100) * (1 + ptcStep.getR() / 100);
+            x *= (1 + rtcStep.getX() / 100) * (1 + ptcStep.getX() / 100);
+            g *= (1 + rtcStep.getG() / 100) * (1 + ptcStep.getG() / 100);
+            b *= (1 + rtcStep.getB() / 100) * (1 + ptcStep.getB() / 100);
+            ratio *= rtcStep.getRho() * ptcStep.getRho();
+            angle += Math.toRadians(ptcStep.getAlpha());
         }
 
-        return getValue(g1, rtcStepValue, ptcStepValue);
-    }
-
-    /**
-     * Get the nominal magnetizing susceptance of a two windings transformer, located on side 2.
-     */
-    public static double getB1(TwoWindingsTransformer twt, boolean twtSplitShuntAdmittance) {
-        return getB1(twt, getCurrentPosition(twt.getRatioTapChanger()), getCurrentPosition(twt.getPhaseTapChanger()), twtSplitShuntAdmittance);
+        return new TapCharacteristics(r, x, g, b, ratio, angle);
     }
 
-    public static double getB1(TwoWindingsTransformer twt, Integer rtcPosition, Integer ptcPosition, boolean twtSplitShuntAdmittance) {
-        return getB1(twtSplitShuntAdmittance ? twt.getB() / 2 : twt.getB(), twt.getRatioTapChanger(), twt.getPhaseTapChanger(), rtcPosition, ptcPosition);
+    public static TapCharacteristics getTapCharacteristics(TwoWindingsTransformer twt, Integer rtcPosition, Integer ptcPosition) {
+        double ratio0 = twt.getRatedU2() / twt.getRatedU1();
+        double angle0 = 0;
+        double r0 = twt.getR();
+        double x0 = twt.getX();
+        double g0 = twt.getG();
+        double b0 = twt.getB();
+        return getTapCharacteristics(twt.getRatioTapChanger(), twt.getPhaseTapChanger(), rtcPosition, ptcPosition, ratio0,
+                angle0, r0, x0, g0, b0);
     }
 
-    /**
-     * Get the nominal magnetizing susceptance of a three windings transformer, located on bus star side.
-     */
-    public static double getB1(ThreeWindingsTransformer.Leg leg, boolean twtSplitShuntAdmittance) {
-        return getB1(leg, getCurrentPosition(leg.getRatioTapChanger()), getCurrentPosition(leg.getPhaseTapChanger()), twtSplitShuntAdmittance);
+    public static Integer getCurrentPosition(RatioTapChanger rtc) {
+        return rtc != null ? rtc.getTapPosition() : null;
     }
 
-    public static double getB1(ThreeWindingsTransformer.Leg leg, Integer rtcPosition, Integer ptcPosition, boolean twtSplitShuntAdmittance) {
-        return getB1(twtSplitShuntAdmittance ? leg.getB() / 2 : leg.getB(), leg.getRatioTapChanger(), leg.getPhaseTapChanger(), rtcPosition, ptcPosition);
+    public static Integer getCurrentPosition(PhaseTapChanger rtc) {
+        return rtc != null ? rtc.getTapPosition() : 0;
```

Comment:
`ptc` should be better. I am not sure to understand why I have initialy chosen `0` when we have no phase tap changer.

Reply:
I renamed to ptc and replace 0 by null to be consistent with the other method.